julia/related.jl

```diff
@@ -9,9 +9,8 @@ function relatedIO()
     json_string = read("../posts.json", String)
     posts = JSON3.read(json_string, Vector{PostData})    
 
-    related(posts)
```

Comment:
Why remove the compilation call and include it in the timing?

Reply:
As the size of the posts is only known during processing (after i/o) any type optimizations have to be measured as well. 
```

function related(posts)
    for T in (UInt8, UInt16, UInt32, UInt64)
        if length(posts) < typemax(T)
            return related(T, posts)
        end
    end
end

```
Is not done during compilation as I understand. It just asserts the type (UInt8, UInt32 etc.) from the posts size.

Reply:
So this "as I understand" of yours somehow translates to the solution of "let's have the full compilation code + 2x runs added to this language result"?

I am trying to be respectful here: this conversation we have here should not occur. And that change you made should never have been done.

Reply:
you're still removing the manual warmup.

This function you're looking at is always called at runtime.

Reply:
What could be done instead is do 4 calls before the timing loop, e.g.
```
related(UInt8, posts[1:20])
related(UInt16, posts[1:20])
related(UInt32, posts[1:20])
related(UInt64, posts[1:20])
```
then every relevant code path should be hit and compiled on a small workload before the timing loop begins.